app/portainer/settings/authentication/internal-auth/PasswordLengthSlider.tsx

```diff
@@ -0,0 +1,102 @@
+import { useEffect, useState } from 'react';
+import RcSlider from 'rc-slider';
+import clsx from 'clsx';
+
+import { Badge } from '@/portainer/components/Badge/Badge';
+
+import 'rc-slider/assets/index.css';
+
+import styles from './PasswordLengthSlider.module.css';
+
+export interface Props {
+  min: number;
+  max: number;
+  step: number;
+  value: number;
+  onChange(value: number): void;
+}
+
+type Strength = 'weak' | 'good' | 'strong' | 'veryStrong';
+type BadgeProperties = { icon: string; color: string; text: string };
+
+const badgeProperties: Record<
+  Strength,
+  { icon: string; color: string; text: string }
+> = {
+  weak: { icon: 'far fa-times-circle', color: 'red', text: 'Weak password' },
+  good: { icon: 'far fa-check-circle', color: 'yellow', text: 'Good password' },
+  strong: {
+    icon: 'far fa-check-circle',
+    color: 'green',
+    text: 'Strong password',
+  },
+  veryStrong: {
+    icon: 'fa fa-lock',
+    color: 'blue',
+    text: 'Very strong password',
+  },
+};
+
+const SliderWithTooltip = RcSlider.createSliderWithTooltip(RcSlider);
+
+export function PasswordLengthSlider({
+  min,
+  max,
+  step,
+  value,
+  onChange,
+}: Props) {
+  const [badgeProps, setBadgeProps] = useState<BadgeProperties>({
+    icon: '',
+    color: '',
+    text: '',
+  });
+
+  function getBadgeProps(value: number) {
+    if (value < 10) {
+      return badgeProperties.weak;
+    }
+
+    if (value < 12) {
+      return badgeProperties.good;
+    }
+
+    if (value < 14) {
+      return badgeProperties.strong;
+    }
+
+    return badgeProperties.veryStrong;
+  }
+
+  useEffect(() => {
+    setBadgeProps(getBadgeProps(value));
+  }, [value]);
+
+  function handleChange(sliderValue: number) {
+    console.log(value);
+    onChange(sliderValue);
+    setBadgeProps(getBadgeProps(sliderValue));
+  }
```

Comment:
badeProps is just a derived state from `value`, so just do it like that:
```suggestion
   const badeProps = getBadeProps(value);

  function getBadgeProps(value: number) {
    if (value < 10) {
      return badgeProperties.weak;
    }

    if (value < 12) {
      return badgeProperties.good;
    }

    if (value < 14) {
      return badgeProperties.strong;
    }

    return badgeProperties.veryStrong;
  }
  function handleChange(sliderValue: number) {
    onChange(sliderValue);
  }
```

Reply:
you're right, initially i added this in the process of debugging something but it was definitely redundant once the onchange method was added to change the value in the parent scope 👍🏻